cedar-policy-core/src/evaluator/err.rs

```diff
@@ -115,25 +116,46 @@ impl EvaluationError {
     }
 
     /// Construct a [`TypeError`] error
-    pub(crate) fn type_error(expected: Vec<Type>, actual: Type) -> Self {
+    pub(crate) fn type_error(expected_first: Type, expected_rest: Vec<Type>, actual: Type) -> Self {
         Self {
-            error_kind: EvaluationErrorKind::TypeError { expected, actual },
+            error_kind: EvaluationErrorKind::TypeError {
+                expected_first,
+                expected_rest,
+                actual,
+            },
             advice: None,
         }
     }
 
+    pub(crate) fn type_error_single(expected_first: Type, actual: Type) -> Self {
```

Comment:
also a comment here on what `expected_first` is (maybe rename to just `expected`?) and what the difference is between this function and `type_error()`.  (It's clear in this PR, but might not be immediately clear to future people coming to this code for the first time)